### ISSUE TITLE
Use gitutils in the `GIT_COMMIT_STATS` syncer

### ISSUE THREAD

**Issue by patrickdevivo** [mergestat/mergestat#407]
Switch to using [`gitutils`](https://github.com/mergestat/gitutils) in the `GIT_COMMIT_STATS` syncer - if it makes sense to do so (it's not signficantly slower).

We probably want to wait on some benchmarks before we invest in the effort.

**Comment by ramiroJCB** [comment]
@patrickdevivo  same question of what feature exactly we should use  from gitutils here ?

**Comment by patrickdevivo** [comment]
See this comment: https://github.com/mergestat/mergestat/issues/406#issuecomment-1311768955 same idea - use the commits iterator (`gitlog` package) to retrieve this data from a repo, instead of `libgit2`

**Comment by patrickdevivo** [comment]
Closing this out as our new container based sync approach makes it no longer relevant. We may or may not use gitutils in that approach in the future (today it continues to use `mergestat-lite`, but as a CLI)
